glam/api/views.py

```diff
@@ -27,6 +30,18 @@
 )
 
 
+class GlamBigQueryClient:
+    """Holds a singleton BigQuery client for GLAM prod."""
+
+    _instance = None
+
+    def __new__(cls):
+        if not cls._instance:
+            cls._instance = super(GlamBigQueryClient, cls).__new__(cls)
+            cls._instance.client = bigquery.Client(project="moz-fx-data-glam-prod-fca7")
+        return cls._instance.client
+
+
```

Comment:
Considering this [gunicorn config](https://github.com/mozilla/glam/blob/e42096fdbc1ad1be2644be462c7aa875ef99da6c/Dockerfile#L82-L84) using `Client` as a context manager seems like a better alternative to ensure sane instance lifecycle.

Reply:
Regarding the singleton, I didn't notice a significant delay and I come to realize it was a mistake. I also didn't know BQ client implemented context manager, so TYVM for this. How does that look now?